addons/sale/static/src/js/sale_product_catalog_quantity.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { IntegerField } from '@web/views/fields/integer/integer_field';
+
+
+export class SaleProductCatalogQuantity extends IntegerField {
+    setup() {
+        super.setup();
+    }
+
+    removeQuantity() {
+        if (this.props.value > 0) {
+            this.props.update(this.props.value - 1);
+        }
+   }
+
+    addQuantity() {
+        this.props.update(this.props.value + 1);
+    }
+
+    // update the quantity whenever the content is changed, otherwise the user
+    // have to click outside of the record to update the SOL quantity
+    onInput(ev) {
+        // otherwise we get a NaN bug when updating values from the keyboard
+        let value = parseInt(ev.target.value)
+        if (value >= 0) {
+            this.props.update(value);
+        }
```

Comment:
strange ?

Reply:
when you clean the input through the keyboard it parses it as NaN and remove the input (as if the content is zero) but this is unwanted for an user that is removing the whole number to put a new one